hcipy/optics/fiber.py

```diff
@@ -158,16 +189,21 @@ def backward(self, instance_data, wavefront):
 
 		return Wavefront(output_electric_field, wavefront.wavelength)
 
+@deprecated('This class operated counter to HCIPy best practices and will be removed in HCIPy 0.7 or later. Please use a combination of the SingleModeFiberInjection and Detector classes instead.')
 class SingleModeFiber(Detector):
 	def __init__(self, input_grid, mode_field_diameter, mode=None):
 		self.input_grid = input_grid
 		self.mode_field_diameter = mode_field_diameter
 
 		if mode is None:
-			mode = gaussian_mode
+			mode = make_gaussian_fiber_mode(mode_field_diameter)
 
-		self.mode = mode(self.input_grid, mode_field_diameter)
-		self.mode /= np.sum(np.abs(self.mode)**2 * self.input_grid.weights)
+		try:
+			self.mode = mode(self.input_grid, mode_field_diameter)
+		except TypeError:
+			self.mode = mode(self.input_grid)
+
+		self.mode /= np.sqrt(np.sum(np.abs(self.mode)**2 * self.input_grid.weights))
 		self.intensity = 0
 
 	def integrate(self, wavefront, dt, weight=1):
```

Comment:
Missing complex conjugate here too.

Reply:
its in the integrate function but I couldn't add a comment there because of ????

Reply:
Because it's not code that is not changed by this PR. I'll put the conjugate there, this time. But because it's a deprecated class, I don't really want to spend time on fixing stuff that's gonna be removed in the near future.
 People should be using the new functions/classes if they run into bugs.

Reply:
Done.